### ISSUE TITLE
RuntimeError - Can't wrap # (arity: -2) to have arity 2 error

### ISSUE THREAD

**Issue by smeriwether** [rmosolgo/graphql-ruby#1173]
Hello! 

GraphQL-Ruby Version: 1.7.7
GraphQL-Batch Version: 0.3.1
Rails Version: 4.2.5.2
Ruby Version: 2.3.4

I updated the gem from version 1.5.9 to version 1.7.7 and I started seeing a `RuntimeError - Can't wrap # (arity: -2) to have arity 2:` error. 

Backtrace:
```
RuntimeError - Can't wrap #<Proc:0x007fe9179ff4e0@/Users/stephenmeriwether/workspace/my-app/app/graph/types/date_time_type.rb:11 (lambda)> (arity: -2) to have arity 2:
  graphql (1.7.7) lib/graphql/backwards_compatibility.rb:29:in `wrap_arity'
  graphql (1.7.7) lib/graphql/scalar_type.rb:108:in `ensure_two_arg'
  graphql (1.7.7) lib/graphql/scalar_type.rb:97:in `coerce_result='
  graphql (1.7.7) lib/graphql/define/instance_definable.rb:293:in `call'
  graphql (1.7.7) lib/graphql/define/defined_object_proxy.rb:38:in `method_missing'
   () Users/stephenmeriwether/workspace/my-app/app/graph/types/date_time_type.rb:11:in `block in <top (required)>'
  graphql (1.7.7) lib/graphql/define/instance_definable.rb:161:in `ensure_defined'
  graphql (1.7.7) lib/graphql/define/instance_definable.rb:204:in `block (3 levels) in stash_dependent_methods'
  graphql (1.7.7) lib/graphql/base_type.rb:95:in `to_s'
  graphql (1.7.7) lib/graphql/base_type.rb:59:in `=='
  graphql (1.7.7) lib/graphql/execution/typecast.rb:8:in `subtype?'
  graphql (1.7.7) lib/graphql/schema/validation.rb:206:in `block (3 levels) in <module:Rules>'
  graphql (1.7.7) lib/graphql/schema/validation.rb:202:in `block (2 levels) in <module:Rules>'
  graphql (1.7.7) lib/graphql/schema/validation.rb:201:in `block in <module:Rules>'
  graphql (1.7.7) lib/graphql/schema/validation.rb:19:in `block in validate'
  graphql (1.7.7) lib/graphql/schema/validation.rb:19:in `validate'
  graphql (1.7.7) lib/graphql/schema/traversal.rb:130:in `validate_type'
  graphql (1.7.7) lib/graphql/schema/traversal.rb:67:in `visit'
  graphql (1.7.7) lib/graphql/schema/traversal.rb:117:in `visit_field_on_type'
  graphql (1.7.7) lib/graphql/schema/traversal.rb:102:in `block in visit_fields'
  graphql (1.7.7) lib/graphql/schema/traversal.rb:101:in `visit_fields'
  graphql (1.7.7) lib/graphql/schema/traversal.rb:72:in `visit'
  graphql (1.7.7) lib/graphql/schema/traversal.rb:53:in `block in visit'
  graphql (1.7.7) lib/graphql/schema/traversal.rb:53:in `visit'
  graphql (1.7.7) lib/graphql/schema/traversal.rb:32:in `initialize'
  graphql (1.7.7) lib/graphql/schema.rb:594:in `rebuild_artifacts'
  graphql (1.7.7) lib/graphql/schema.rb:237:in `types'
  graphql (1.7.7) lib/graphql/schema/validation.rb:133:in `block in <module:Rules>'
  graphql (1.7.7) lib/graphql/schema/validation.rb:19:in `block in validate'
  graphql (1.7.7) lib/graphql/schema/validation.rb:19:in `validate'
  graphql (1.7.7) lib/graphql/schema.rb:198:in `define'
```

DateTimeType Definition:
```
DateTimeType = GraphQL::ScalarType.define do
  name 'DateTime'
  description 'Date with time in ISO 8601'

  coerce_input lambda { |value, *|
    return value if value.blank?

    Time.zone.parse(value.to_s)
  }

  coerce_result lambda { |value, *|
    if value.is_a? String
      Time.parse(value).iso8601
    else
      value.iso8601
    end
  }
end
```

Any help would be greatly appreciated! Thanks! #

**Comment by smeriwether** [comment]
I *think* this issue might be related? https://github.com/rmosolgo/graphql-ruby/issues/979

That seems to be fixed in a pre-release version of this gem. The primary reason for updating is to take advantage of GraphQL subscription support (thanks a ton for adding that!). Another temporary solution for me might be to update the gem to a version that includes subscription support and still supports my DateTimeType (I would need some help figuring out what that version might be)? Just a thought.

**Comment by smeriwether** [comment]
I found a fix for this, it was by updating the function definition from `lambda { |value, ctx|` to `->(value, ctx)`

The reason being this:
```
irb(main):024:0> foo = ->(a, b) { puts a, b }
=> #<Proc:0x007fca88102688@(irb):24 (lambda)>
irb(main):025:0> bar = lambda { |value, *| puts value }
=> #<Proc:0x007fca888c85c0@(irb):25 (lambda)>
irb(main):026:0> foo.arity
=> 2
irb(main):027:0> bar.arity
=> -2
irb(main):028:0>
```

**Comment by rmosolgo** [comment]
Thanks for the great bug report and I'm glad you found a work-around! 

This is fixed in 1.8.0, which is available as a pre-release, see https://github.com/rmosolgo/graphql-ruby/issues/979#issuecomment-344311998
